### ISSUE TITLE
Support cabal-install >= 2.4.1.0 in the Makefile

### ISSUE THREAD

**Issue by asr** [agda/agda#3247]
Using that version of `cabal-install`I'm getting the following warning:

```bash
$ make install-bin
...
Warning: The install command is a part of the legacy v1 style of cabal usage.

Please switch to using either the new project style and the new-install
command or the legacy v1-install alias as new-style projects will become the
default in the next version of cabal-install. Please file a bug if you cannot
replicate a working v1- use case with the new-style commands.
```

I created this issue for recording the changes required.

**Comment by UlfNorell** [comment]
Using `new-install` doesn't work at the moment due (at least) to https://github.com/haskell/cabal/issues/5550.

**Comment by fgaz** [comment]
5550 will be fixed in the next cabal-install point release

**Comment by UlfNorell** [comment]
It looks like `cabal-2.4.1.0` might work. At least the `build-dir` issue is fixed.

**Comment by asr** [comment]
Cabal install 2.4.1.0 requires Cabal library >=2.4.1.0 && <2.5, but GHC 8.6.3 is shipped with Cabal library 2.4.0.1. I'll wait for a version of GHC shipped with a version of Cabal library required by Cabal install 2.4.1.0 for fixing this issue.

**Comment by asr** [comment]
@phadej [wrote](https://github.com/agda/agda/issues/3369#issuecomment-463297069):
 
Note: https://github.com/haskell/cabal/issues/5889

use of `die` in `Setup.hs` will prevent people from `cabal v2-install` the package.

This is easy to reproduce: in `release-2.5.4.3` or `master` branches, using `cabal-install-2.4` `cabal v2-build` will fail with

```haskell
Generating Agda library interface files...
/home/ogre/.cabal/share/x86_64-linux-ghc-8.4.4/Agda-2.5.4.2.20190111/lib/prim/Agda/Primitive.agda:1,1-1
Cannot read file /home/ogre/.cabal/share/x86_64-linux-ghc-8.4.4/Agda-2.5.4.2.20190111/lib/prim/Agda/Primitive.agda
Error: /home/ogre/.cabal/share/x86_64-linux-ghc-8.4.4/Agda-2.5.4.2.20190111/lib/prim/Agda/Primitive.agda: openFile: does not exist (No such file or directory)
Error: Failed to typecheck /home/ogre/.cabal/share/x86_64-linux-ghc-8.4.4/Agda-2.5.4.2.20190111/lib/prim/Agda/Primitive.agda!
cabal: Failed to build Agda-2.5.4.2.20190111. The failure occurred during the
final install step.
```

i.e. one cannot use `cabal v2-build` to even develop `Agda`.

**Comment by asr** [comment]
Fixed by 378020321a223d7057cd10b59b995ebac91c7a8b which was tested with `cabal-2.4.1.0` and all the Cabal install versions used in Travis. Thanks @phadej.
